hugophotoswipe/photo.py

```diff
@@ -105,13 +108,105 @@ def _load_original_image(self):
+
+        exif_data = {}
+        exif = image._getexif() or {}
+        for k, v in exif.items():
+            decoded = TAGS.get(k)
+            if decoded in tags and not isinstance(v, IFDRational):  # Filter complex data values
+                exif_data[decoded] = v
+        for k, v in exif_data.pop('GPSInfo', {}).items():
+            decoded = GPSTAGS.get(k, k)
+            exif_data[decoded] = v
+        self._exif = exif_data
+
     def free(self):
         """Manually clean up the cached image"""
         if hasattr(self, "_original_img") and self._original_img:
             self._original_img.close()
             del self._original_img
         self._original_img = None
 
+    @property
+    def iptc(self):
+        if self._iptc is None:
+            if settings.iptc:
+                tags = _filter_tags(iptc3.c_datasets_r.keys(),
+                                    settings.iptc.get('include'),
+                                    settings.iptc.get('exclude'))
+            else:
+                tags = iptc3.c_datasets_r.keys()
+
+            info = iptc3.IPTCInfo(self.original_path)
+            iptc = {}
+            for k in tags:
+                if type(info[k]) is bytes:
+                    iptc[k] = info[k].decode('utf-8')
+                elif type(info[k]) is list:
+                    l = []
+                    for v in info[k]:
+                        l.append(v.decode('utf-8'))
+                    iptc[k] = l
+                else:
+                    iptc[k] = info[k]
+            self._iptc = iptc
+
+        return self._iptc
+
+    @property
+    def exif(self):
+        if not self._exif:
+            _ = self.original_image  # Trigger loading image and exif data
+        return self._exif
+
+    def _get_tag_value(self, tag):
+        assert tag is not None
+        try:
+            obj, t = tag.split('.')
+        except Exception as e:
+            logging.warning(e)
+            raise ValueError(f"Tag improperly formatted. Should be of format (exif/iptc).tag Provided: ({tag})")
+        if obj.lower() not in ['exif', 'iptc']:
+            raise ValueError(f"Tags can only reference iptc or exif data. ({tag})")
+        o = getattr(self, obj.lower(), {})
+        if o is None:
+            logging.warning(f'Tag "{tag}" specified but {obj} not loaded. Returning "".')
+            o = {}
+        return o.get(t, "")
+
+    @property
+    def caption(self):
+        if self._caption:
+            return self._caption
+        elif settings.tag_map and settings.tag_map.get('caption'):
+            return str(self._get_tag_value(settings.tag_map.get('caption')))
+        return ""
+
+    @property
+    def copyright(self):
+        if self._copyright:
+            return self._copyright
+        elif settings.tag_map and settings.tag_map.get('copyright'):
+            return str(self._get_tag_value(settings.tag_map.get('copyright')))
+        return ""
+
+    def __getattribute__(self, attr):
```

Comment:
I'm not a huge fan of overloading `__getattribute__` like this, especially because you're catching an `AttributeError` and then looking in the settings map. What if in the future I unintentionally introduce a class variable that's also an exif tag name? I'm not entirely sure where you're using this, but perhaps a `get_tag` method can work instead? Happy to discuss this if you'd like.

Reply:
The reason for overloading `__getattribute__` is so that you can access those tags from within the hugo shortcode. That would allow you to expand the shortcode with, for example, `{{ .Get "aperture"}}`. I couldn't figure out another way to make that data accessible when processing the shortcode, except for explicitly defining each as a property. I'm definitely open to other ideas.

I do understand your objection though and as it isn't quite immediately related to the exif/iptc feature I'm removing this from this branch until we can find out a better solution.

Reply:
Sorry for the slow response!

This makes sense, Here's one solution I can think of: require a prefix when used in a shortcode: `{{ .Get "exit:ApertureValue" }}` or `{{ .Get "iptc:ReleaseTime" }}`. Then you can overload the `__getattribute__` method by first checking for the `exif` or `iptc` prefixes, and retrieving them using `_get_tag_value`. If they don't start with either prefix, return `getattr(self, attr)`.